ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -370,8 +416,10 @@ private void registerTransientQuery(
       final MetaStore metaStore,
       final TransientQueryMetadata query
   ) {
-    // Initialize the query before it's exposed to other threads via {@link allLiveQueries}.
-    query.initialize();
+    if (!query.isInitialized()) {
+      throw new IllegalStateException("Transient query must be initialized before it might"
+          + " be exposed to other threads via allLiveQueries");
+    }
```

Comment:
We just needed to initialize before we register, it wasn't a requirement to actually do it in the same method.